--- external/jsoncpp_ORB/Android.bp	2022-05-18 15:14:44.703072013 +0100
+++ Androidbp	2022-05-18 15:16:44.376193765 +0100
diff --git a/Android.bp b/Android.bp
index 03e8ac1..db60a13 100644
--- a/Android.bp
+++ b/Android.bp
@@ -29,7 +29,7 @@ license {
 }
 
 cc_defaults {
-    name: "libjsoncpp_defaults",
+    name: "libjsoncpp_ORB_defaults",
 
     srcs: [
         "src/lib_json/json_reader.cpp",
@@ -62,35 +62,17 @@ cc_defaults {
 }
 
 cc_library {
-    name: "libjsoncpp",
-    defaults: ["libjsoncpp_defaults"],
-    vendor_available: true,
-    product_available: true,
-    vndk: {
-        enabled: true,
-        support_system_process: true,
-    },
+    name: "libjsoncpp_ORB",
+    defaults: ["libjsoncpp_ORB_defaults"],
     host_supported: true,
-    ramdisk_available: true,
-    vendor_ramdisk_available: true,
+    ramdisk_available: true,    
     recovery_available: true,
-    apex_available: [
-        "//apex_available:platform",
-        "//apex_available:anyapex",
-    ],
     native_bridge_supported: true,
     min_sdk_version: "29",
 }
 
-cc_library_static {
-    name: "libjsoncpp_ndk",
-    defaults: ["libjsoncpp_defaults"],
-    sdk_version: "9",
-    stl: "libc++_static",
-}
-
 cc_library_headers {
-    name: "libjsoncpp_headers",
+    name: "libjsoncpp_ORB_headers",
     export_include_dirs: ["include"],
     host_supported: true,
 }
