diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index c285f8cd5d30..306457b01031 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -206,7 +206,7 @@
     },
     {
       name: "AudioVideoTracks",
-      status: "experimental",
+      status: "stable",
     },
     {
       name: "AutoLazyLoadOnReloads",
